### PR TITLE
fix: replace actor heartbeat duration with emit timeout as the loop i…

### PR DESCRIPTION
**Pull request by etolbakov**
### Description

replace actor heartbeat duration with emit timeout as the loop interval for sources.

### How was this PR tested?
This is a configuration change, no special tests have been made

### Related issues
https://github.com/quickwit-oss/quickwit/issues/3829

### CODE REVIEW

File: quickwit/quickwit-indexing/src/source/kafka_source.rs [hunk 1]
```diff
@@ -49,7 +49,10 @@ use tracing::{debug, info, warn};
...
 use crate::actors::DocProcessor;
 use crate::models::{NewPublishLock, PublishLock};
-use crate::source::{BatchBuilder, Source, SourceContext, SourceRuntimeArgs, TypedSourceFactory};
+use crate::source::{
+    BatchBuilder, Source, SourceContext, SourceRuntimeArgs, TypedSourceFactory,
+    EMIT_BATCHES_TIMEOUT,
+};
 
 /// Number of bytes after which we cut a new batch.
 ///
```
Comment (etolbakov) [inline quickwit/quickwit-indexing/src/source/kafka_source.rs:58]: I've noticed that here and in other sources we are using something like `BATCH_NUM_BYTES_LIMIT: u64 = 5_000_000`.
1. I'm happy to replace it with Bytsize
2. should it be `5Mb` (as it here) or `5Mib`(like in [mod.rs](https://github.com/quickwit-oss/quickwit/pull/4112/files#diff-a2df65a154da2a6ab79da9866d7a5dfca567d9951ee3bd3eb2687fe3d4681801))?

**Comment by etolbakov** [comment]
Adrien could you please take a look when you have time?

File: quickwit/quickwit-indexing/src/source/kafka_source.rs (hunk 1 above)
Comment (guilload) [reply quickwit/quickwit-indexing/src/source/kafka_source.rs:58]: 1. Yes, good idea. Let's use a shared `Bytsize ` constant defined in the source module for all sources, as you suggested.

2. Up to you :)

**Review by guilload — APPROVED** [review]

**Comment by etolbakov** [comment]
Adrien I've made some changes (hopefully improvements 😅) could you please take a look when you have a spare minute?

**Comment by guilload** [comment]
Once the PR is approved, feel free to get the branch in a mergeable state and merge it yourself. You should have access. Let me know if it's not the case.

**Comment by guilload** [comment]
Merged. Thanks a lot @etolbakov !

**Pull request merged**